hydrolib/core/basemodel.py

```diff
@@ -523,6 +563,20 @@ def _resolved_filepath(self) -> Optional[Path]:
         with file_load_context() as context:
             return context.resolve(self.filepath)
 
+    @property
+    def save_location(self) -> Path:
+        """Gets the current save location which will be used when calling `save()`
+
+        Returns:
+            Path: The location at which this model will be saved.
+        """
+        filepath = self.filepath or self._generate_name()
```

Comment:
Nitpick, this behaviour is a duplicate of the actual save functionality. If you cannot easily change this, maybe add a comment in the save function to always update this one as well.

Reply:
It is not really a duplicate, we can call the `save_location` from a model created in memory, in which case it seems weird to change state when calling a property. I'd honestly rather have a default value for filepath, and not have it optional. This way we do not have to worry about it being potentially `None` anymore. 

If you agree I will create a follow up issue to do this.